apps/src/lib/ui/SettingsCog.jsx

```diff
@@ -117,42 +106,40 @@ export default class SettingsCog extends Component {
     const {isRunning, runModeIndicators} = this.props;
 
     return (
-      <span
-        className={classNames(
-          moduleStyles.iconContainer,
-          runModeIndicators && isRunning && moduleStyles.iconContainerRunning
-        )}
-        ref={icon => this.setTargetPoint(icon)}
-      >
+      <span>
         <button
           type="button"
-          onClick={this.open}
-          className={moduleStyles.button}
+          onClick={this.toggleOpen}
+          className={classNames(
+            moduleStyles.settingsButton,
+            runModeIndicators && isRunning && moduleStyles.settingsButtonRunning
+          )}
         >
           <FontAwesome
             className="settings-cog"
             icon="cog"
             title={msg.settings()}
           />
         </button>
-        <PopUpMenu
-          className="settings-cog-menu"
-          targetPoint={this.targetPoint}
-          isOpen={this.state.open}
-          onClose={this.close}
-          showTail={true}
-        >
-          <ManageAssets onClick={this.manageAssets} />
-          {this.areLibrariesEnabled() && (
-            <ManageLibraries onClick={this.manageLibraries} />
-          )}
-          {this.areAIToolsEnabled() && (
-            <ManageModels onClick={this.manageModels} />
-          )}
-          {this.props.showMakerToggle && (
-            <ToggleMaker onClick={this.toggleMakerToolkit} />
-          )}
-        </PopUpMenu>
+        {this.state.open && (
+          <JavalabDropdown>
```

Comment:
may be better to do as a follow-up, but should we move and rename `JavalabDropdown` so it is clear it is a reusable component? It's a little confusing to have it here

Reply:
Yeah good point, I think an immediate follow-up makes sense. Any recollection of where we theorized reusable components should go during our apps organization discussion at in-person week?

Reply:
I wasn't in that meeting, but per the [doc](https://docs.google.com/document/d/1IVSopUi7e4Wu_87o2GFZYpTJqs73E755V9Cxq2_3Moc/edit#heading=h.8ii0vz5oblwl) it doesn't look like any conclusions were made?

Reply:
d'oh, thanks! will review

Reply:
Thanks for sharing the doc @molly-moen. I enjoyed browsing through [refactor.md](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/docs/refactor.md) that @madelynkasula shared. Wonder if this component would belong in either the mysterious `templates` directory or maybe in `lib/ui` where SettingsCog lives?

Reply:
Maybe `apps/src/componentLibrary`? I don't think the JavalabDropdown component is thaaat great right now, but maybe if we put it there we could keep improving it / start using it in more places?

Reply:
`apps/src/componentLibrary` would make sense based on naming. We could follow up on the discussion captured by [this doc](https://docs.google.com/document/d/1IVSopUi7e4Wu_87o2GFZYpTJqs73E755V9Cxq2_3Moc/edit#) and provide some kind of guidance/documentation on how to determine which directory a reusable component would be stored.